### ISSUE TITLE
.NET Core Support

### ISSUE THREAD

**Issue by robertmuehsig** [OfficeDev/Open-XML-SDK#65]
I just tried the .NET Portability Analyzer on the current and the vNext branch and it seems there are only a few code parts that needs to be changed to run it on .NET Core:

![image](https://cloud.githubusercontent.com/assets/756703/11028594/160756d6-86c0-11e5-8822-aff91e214b06.png)

Have you any plans to support it?

Is your System.IO.Packaging the same code as in the CoreFX (which would run under .NET Core) repository or is this something "OpenXml" specific?  https://github.com/dotnet/corefx/tree/master/src/System.IO.Packaging

**Comment by EricWhiteDev** [comment]
The System.IO.Packaging is the same - I wrote the new System.IO.Packaging such that it would work in .NET Core, and submitted it to COREFX, along with a different set of XUnit tests, so we are good-to-go on that front.

The Open XML SDK uses, in a few places, some deprecated classes (XmlTextReader IIRC) that are not in COREFX.  It should be trivial to convert that code to use supported COREFX classes.

The big task in porting to COREFX is finishing some additional XUnit tests, so that we can validate functionality.  I am currently in the process of madly writing XUnit tests.  When I get to a point where running the XUnit tests gives us a good idea that the functionality of the SDK is good, then doing the port to COREFX and running the XUnit tests will be an easy task (I hope, but one never knows).

I'm assigning to myself, but don't expect anything for a month or two.  I am on vacation for another week, and then expect to spend the following two weeks writing XUnit tests.  Following that I am slated to do the port to Office 2016.

**Comment by JaredShaver** [comment]
Since the .NET Core ecosystem heavily relies on NuGet packages will issue #2 be addressed as part of this?

**Comment by EricWhiteDev** [comment]
@JaredShaver It is possible that issue #2 will be addressed as part of this.  It depends on how much extra work will be involved.

**Comment by adrien-constant** [comment]
I would really be interested in having the support for .NET Core.
My project had to drop dnx50 because of this.

**Comment by wmccullough** [comment]
Just curious to see how this is going. I'm in the same boat as @adrien-constant

**Comment by kwaazaar** [comment]
.net core would be a GREAT plus, since it supports running our solutions on Mac, Linux and even Docker! It may even enhance performance, since .NET core makes some promises on that as well. :)

**Comment by timbophillips** [comment]
+1 from me on this one

**Comment by xrkolovos** [comment]
now that we have the new NET Platform Standard system and .net core rc2 might be the time for this features.. :) 
+1

**Comment by marspd** [comment]
Any progress on porting to .Net Core at all
+1

**Comment by xrkolovos** [comment]
rerun portability report
![image](https://cloud.githubusercontent.com/assets/517802/16359481/8adf12c2-3b3e-11e6-9db8-3c0677fe1558.png)

**Comment by DavidMoraisFerreira** [comment]
Any updates?

**Comment by Supratentorial** [comment]
Any update on this?

**Comment by xrkolovos** [comment]
Last night i created this. I tried out before for our project and it works fine.
[Open XML SDK for NET Platform Standard 3 & .net Core](https://github.com/xrkolovos/Open-XML-SDK-for-NET-Platform-Standard)

It's not proper code, but a first try to help this community. If someone wants to help i can add him to this repo.

**Comment by xrkolovos** [comment]
this might help more -> https://github.com/dotnet/corefx/issues/9635

**Comment by ph1ll** [comment]
I am currently working on full .NET Core support under https://github.com/ph1ll/Open-XML-SDK/tree/netstandard. Please ignore the commit history as this is very much WIP and it is a bit of a mess, obviously there is a lot of work to do and I have been committing as I go along so I can easily revert if I break something. I will be rebasing before submitting a MR. 

Main changes I had to make are:

 * Switching to `dotnet` style builds for simple and cross platform builds/tests
 * Update framework calls where required, mainly around changes in `System.Reflection`
 * Replace `BinaryFormatter` (I've had to add a custom tool to convert the schemas as they are generated outside of this repo, see #101 )
 * Updating unit tests so they are simpler. E.g. using xUnit `Theory` and can be easily called with `dotnet test`

Current status is, I have the main library building, and ~400 unit tests moved across. I have a few failing tests which I need to address. I will consider it complete when all tests are ported and working. 

I would be keen to know the current state of this repo as development seems to have dried up since March. Ideally it would be good to see .NET Core support in this repo and official NuGet packages.

**Comment by davidnmbond** [comment]
Hi All - I'm keen to get this up as a proper nuget package and am willing to contribute to a project which aims to do so.  

xrkolovos, ph1ll,
Please let me know if I can help here.  I have several packages up on Nuget already, so understand the general process here.

**Comment by xrkolovos** [comment]
If anyone wants to add him to my repo and help. I welcome him/her. Although the code is not ready i thing to add it in a nuget package. Better wait for an official one.

**Comment by twsouthwick** [comment]
Work for this is being tracked by #111

**Comment by xrkolovos** [comment]
help me understand because i am confused with all this issues about the same issue. Issue #111 is closed but this issue(#65) is still opened. What are the issues right now?
Thanks

**Comment by tomjebo** [comment]
@xrkolovos The porting for .Net Core is done and yet we are still working on the Nuget packaging. We are working on clearing up the issues list to take care of ones invalidated by the port.

**Comment by MakanMakan** [comment]
Any news ?

**Comment by twsouthwick** [comment]
@thyams @MakanMakan @xrkolovos There is now a nuget package available. It's a CI build (not supported for production use at the moment), but should get you unblocked to try things out. Please open new issues for any problems encountered.

**Comment by twsouthwick** [comment]
See the README for directions on the feed to use

**Comment by thyams** [comment]
I implemented a generic spreadsheet. Submitting issue found as a new issue.

**Comment by tomjebo** [comment]
Closing

**Comment by Bridgeway** [comment]
Package 'SpreadsheetLight 3.4.9' was restored using '.NETFramework, Version=v4.6.1' instead of the project target framework '.NETCoreApp, Version=v2.1'

Given the above message after I installed from Nuget Gallery, assuming I don't run into any errors, is it still feasable to use this package in production release ?

Could I install the dll directly in the output folder (to stop the NuGet folder complaining ?)

**Comment by GeorgDangl** [comment]
As far as I know, SpreadsheetLight does not support .NET Core (or .NET Standard).
I've also used this before, but now have switched to [ClosedXML](https://github.com/ClosedXML/ClosedXML) and I'm quite happy with it.

**Comment by Bridgeway** [comment]
So .NET core 2.1 doesn't have any equivalent to
System.Threading.Tasks;
System.Security.Cryptography;
System.IO;

???

**Comment by twsouthwick** [comment]
@Bridgeway For your first question, v4.6.1 was restored because there is not a .NET Standard or .NET Core version available. .NET Core 2.x supports .NET Framework libraries; however, if an API does not exist, it will throw a `MissingMethodException`. This allows older libraries that do not have modern support to be able to run on newer runtimes - you are just on the hook to test to ensure any paths that use unavailable APIs are known. You can use [ApiPort](https://github.com/microsoft/dotnet-apiport) to see if the library has APIs not available on .NET Core.

> So .NET core 2.1 doesn't have any equivalent to
> System.Threading.Tasks;
> System.Security.Cryptography;
> System.IO;
>  ???

.NET Core 2.1 does support these (see http://apisof.net). The better question would be for specific APIs (ie classes or even members). For that, again, I'd suggest using ApiPort.

**Comment by twsouthwick** [comment]
Also, feel free to open a new issue if you encounter any with the SDK. This issues has been closed for a while and it's nice to keep each issue to its own thread.
